src/meta/src/manager/sink_coordination/coordinator_worker.rs

```diff
@@ -523,24 +535,168 @@ impl CoordinatorWorker {
                     metadatas.push(metadata);
                 }
 
-                let start_time = Instant::now();
-                run_future_with_periodic_fn(
-                    coordinator.commit(epoch, metadatas, first_add_columns),
-                    Duration::from_secs(5),
-                    || {
-                        warn!(
-                            elapsed = ?start_time.elapsed(),
-                            %sink_id,
-                            "committing"
-                        );
-                    },
-                )
-                .await
-                .map_err(|e| anyhow!(e))?;
-                self.handle_manager
-                    .ack_commit(epoch, commit_requests.handle_ids)?;
+                match &mut coordinator {
+                    SinkCommitCoordinator::SinglePhase(coordinator) => {
+                        let start_time = Instant::now();
+                        run_future_with_periodic_fn(
+                            coordinator.commit_directly(epoch, metadatas, first_add_columns),
+                            Duration::from_secs(5),
+                            || {
+                                warn!(
+                                    elapsed = ?start_time.elapsed(),
+                                    %sink_id,
+                                    "committing"
+                                );
+                            },
+                        )
+                        .await
+                        .map_err(|e| anyhow!(e))?;
+                        self.handle_manager
+                            .ack_commit(epoch, commit_requests.handle_ids)?;
+                    }
+                    SinkCommitCoordinator::TwoPhase(coordinator) => {
+                        let commit_metadata = coordinator
+                            .pre_commit(epoch, metadatas, first_add_columns)
+                            .await?;
+                        persist_pre_commit_metadata(
+                            &db,
+                            sink_id as _,
+                            epoch,
+                            commit_metadata.clone(),
+                        )
+                        .await?;
+                        self.handle_manager
+                            .ack_commit(epoch, commit_requests.handle_ids)?;
+                        // Get the latest hummock_committed_epoch and the receiver
+                        let (hummock_committed_epoch, mut rw_futures_utilrx) =
```

Comment:
It's expected to call `subscriber` for only once and store the rx, and then received newly committed epoch from the rx.